core/include/detray/core/detector.hpp

```diff
@@ -444,6 +446,58 @@ class detector {
         return _surfaces_finder;
     }
 
+    /** Produce navigation candidates for a track in a given volume by calling
+     * the volumes surface finder.
+     *
+     * @param vol the reference volume
+     * @param track the track state
+     *
+     * @return a collection of navigation candidates
+     */
+    template <typename track_t>
+    DETRAY_HOST_DEVICE
+    inline vector_type<intersection> get_candidates(
```

Comment:
Hmm... I have to say I prefer the original `navigator.hpp`. What is the exact motivation for this?

Reply:
No particular one, only to keep the interfaces better defined. I can change it back

Reply:
Yeah. By the way, we always have to avoid `vector_type` output which doesn't work in device code even with vecmem

Reply:
Yes, that's what we discussed in the meeting, right?

Reply:
Kind of. I am trying to send `vecmem::vector_buffer` of candidates to device because we cannot create `vector_type` directly in device. `vector_buffer` can be converted into `device_vector` where we can use `push_back()`.

Reply:
But do we need a full vector type? Sorry, if you have to explain this to me a bit, I am just trying to understand the problem :)

Reply:
> But do we need a full vector type? 

Do you mean using vector type for candidates?

Reply:
I mean, we need to somehow iterate and and sort through a piece of dynamic memory, if I get this right? Maybe we can use something less sophisticated than a vector?

Reply:
Right - But is there anything less sophisticated than a vector?

Reply:
A direct memory allocation? OK, that's probably a bad idea. Forget that I said anything :)

Reply:
If you have array in mind... hmm

Reply:
Yeah I want to stick to the current design of `navigator.hpp` unless I come across really bad issues

Reply:
No, problem. I'll change that later